CMakeLists.txt

```diff
@@ -7,10 +7,16 @@ set(CMAKE_CXX_EXTENSIONS OFF)
 
 set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${CMAKE_SOURCE_DIR}/cmake/Modules/")
 
-include(GetGitRevisionDescription)
-git_describe(PROJECT_VERSION --tags)
-if(NOT PROJECT_VERSION)
-	set(PROJECT_VERSION "1.2+")
+if(NOT PERFORMOUS_VERSION)
+    message("No project version specified. Determining latest tag..")
+    include(GetGitRevisionDescription)
+    git_describe(PROJECT_VERSION --tags)
+    if(NOT PROJECT_VERSION)
+        message("No project version found. Falling back onto default versioning strategy.")
+        set(PROJECT_VERSION "1.2+")
+    endif()
+else()
+    set(PROJECT_VERSION "${PERFORMOUS_VERSION}")
```

Comment:
maybe we could call the project() here after so that we can pass the correct version to it (project sets a full number of other variables (as major, minor patch tweak)).

Reply:
According to cmake's own documentation this is already done by supplying a value for `PROJECT_VERSION`: 

> [PROJECT_VERSION](https://cmake.org/cmake/help/latest/variable/PROJECT_VERSION.html#project-version)
Value given to the VERSION option of the most recent call to the [project()](https://cmake.org/cmake/help/latest/command/project.html#command:project) command, if any.

Reply:
yes, but as you did not pass it to project, I guess all will be empty

Reply:
That's not how i read this but i might be wrong heheh. I'd say we take it to the test, and check Major, Minor, Patch and Tweak by printing them to the console with Message. I won't be able to do this until way later today

Reply:
Seems this won't even work: 
```
CMake Error at C:\Users\Arjan\source\repos\performous\performous2\CMakeLists.txt:25 (project):
  VERSION "1.2.0-10-54es68d-alpha" format invalid.	Performous	C:\Users\Arjan\source\repos\performous\performous2\CMakeLists.txt	25	
```

This is because cmake only supports the format `Major.Minor.Patch.Tweak`. But they all need to be postive integers.
Keeping this as it is right now